tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/ExtensionMethodsSuite.scala

```diff
@@ -0,0 +1,366 @@
+        Nil,
+        tparam("x", "String"),
+        tname("<"),
+        Nil,
+        List(List(tparam("y", "String"))),
+        None,
+        int(2)
+      )
+    )
+    runTestAssert[Stat]("def (x: String) +:(y: String) = 2")(
+      Defn.ExtensionMethodInfix(
+        Nil,
+        tparam("x", "String"),
+        tname("+:"),
+        Nil,
+        List(List(tparam("y", "String"))),
+        None,
+        int(2)
+      )
+    )
+  }
+
+  // ---------------------------------
+  // EXTENSION GROUP
+  // ---------------------------------
+
+  test("extension-anonymous") {
+    runTestAssert[Stat](
+      "extension {\n  def (c1: Circle).cf1: Int = 2\n  def (c2: Circle).cf2: Int = 2\n}"
+    )(
+      Defn.ExtensionGroup(
+        Nil,
+        anon,
+        Nil,
+        Nil,
+        Term.Param(Nil, anon, None, None),
+        tpl(
+          List(
+            circleExtMethod("c1", "cf1"),
+            circleExtMethod("c2", "cf2")
+          )
+        )
+      )
+    )
+  }
+
+  test("extension-named") {
+    runTestAssert[Stat](
+      "extension ext {\n  def (c1: Circle).cf1: Int = 2\n  def (c2: Circle).cf2: Int = 2\n}"
+    )(
+      Defn.ExtensionGroup(
+        Nil,
+        pname("ext"),
+        Nil,
+        Nil,
+        Term.Param(Nil, anon, None, None),
+        tpl(
+          List(
+            circleExtMethod("c1", "cf1"),
+            circleExtMethod("c2", "cf2")
+          )
+        )
+      )
+    )
+  }
+
+  test("extension-on") {
+    runTestAssert[Stat]("extension on (c: Circle) {\n  def cf1: Int = 2\n  def cf2: Int = 2\n}")(
+      Defn.ExtensionGroup(
+        Nil,
+        anon,
+        Nil,
+        Nil,
+        tparam("c", "Circle"),
+        tpl(
+          List(
+            circleAnonExtMethod("cf1"),
+            circleAnonExtMethod("cf2")
+          )
+        )
+      )
+    )
+    runTestAssert[Stat](
+      "extension ext on (c: Circle) {\n  def cf1: Int = 2\n  def cf2: Int = 2\n}"
+    )(
+      Defn.ExtensionGroup(
+        Nil,
+        pname("ext"),
+        Nil,
+        Nil,
+        tparam("c", "Circle"),
+        tpl(
+          List(
+            circleAnonExtMethod("cf1"),
+            circleAnonExtMethod("cf2")
+          )
+        )
+      )
+    )
+  }
+
+  test("extension-generis-group") {
+    val tTpe = Type.Param(Nil, pname("T"), Nil, Type.Bounds(None, None), Nil, List(pname("Ord")))
+    runTestAssert[Stat]("extension ext on [T: Ord](c: Circle) { def f: Int = 2 }")(
+      Defn.ExtensionGroup(
+        Nil,
+        pname("ext"),
+        List(tTpe),
+        Nil,
+        tparam("c", "Circle"),
+        tpl(
+          List(
+            circleAnonExtMethod("f")
+          )
+        )
+      )
+    )
+  }
+
+  test("extension-negative") {
+    runTestError("extension { }", "Anonymous instance must have at least one extension method")
+    runTestError(
+      "extension on (s: String) { }",
+      "Anonymous instance must have at least one extension method"
```

Comment:
I wonder if we need this check? Especially in a case that for example when we use the parser in Metals and jsut start writing the extension group